shell/browser/ui/file_dialog_gtk.cc

```diff
@@ -66,30 +155,42 @@ class FileChooserDialog {
     else if (action == GTK_FILE_CHOOSER_ACTION_OPEN)
       confirm_text = gtk_util::kOpenLabel;
 
-    dialog_ = gtk_file_chooser_dialog_new(
-        settings.title.c_str(), nullptr, action, gtk_util::kCancelLabel,
-        GTK_RESPONSE_CANCEL, confirm_text, GTK_RESPONSE_ACCEPT, NULL);
+    InitGtkFileChooserNativeSupport();
+
+    if (supports_gtk_file_chooser_native.value_or(false)) {
```

Comment:
```suggestion
    if (*supports_gtk_file_chooser_native) {
```

(here and below) I might be misreading but don't see any code path where the fallback value in `value_or(fallback_value)` will ever be  reached?

Reply:
I think you are right. I will get to this change after work.